src/snowflake/cli/plugins/snowpark/common.py

```diff
@@ -197,3 +203,25 @@ def format_arg(arg: Argument):
         schema=udf_sproc.schema_name,
     )
     return f"{name}({arguments})"
+
+
+def _compare_imports(
+    resource_json: dict, imports: List[str], artifact_file: str
+) -> bool:
+    pattern = re.compile(r"(?:\[@?\w+_\w+\.)?(\w+(?:/\w+)+\.\w+)(?:\])?")
```

Comment:
I suggest to move this to the top of the file.
In this case it does not matter, but it is a good practice to compile regex only once instead on every function call.